src/sql/src/plan/statement/ddl.rs

```diff
@@ -2623,52 +2642,80 @@ impl TryFrom<&KafkaConnectionOptionExtracted> for Option<KafkaSecurity> {
     }
 }
 
-impl TryFrom<KafkaConnectionOptionExtracted> for KafkaConnection {
-    type Error = PlanError;
-    fn try_from(value: KafkaConnectionOptionExtracted) -> Result<Self, Self::Error> {
+impl KafkaConnectionOptionExtracted {
+    fn to_connection(
+        self,
+        scx: &StatementContext,
+    ) -> Result<mz_storage::types::connections::KafkaConnection, PlanError> {
         Ok(KafkaConnection {
-            brokers: value.get_brokers()?,
-            security: (&value).try_into()?,
-            progress_topic: value.progress_topic,
+            brokers: self.get_brokers(scx)?,
+            security: Option::<KafkaSecurity>::try_from(&self)?,
+            progress_topic: self.progress_topic,
         })
     }
 }
 
 generate_extracted_config!(
     CsrConnectionOption,
+    (AwsPrivatelink, with_options::Object),
+    (Port, u16),
     (Url, String),
     (SslKey, with_options::Secret),
     (SslCertificate, StringOrSecret),
     (SslCertificateAuthority, StringOrSecret),
     (Username, StringOrSecret),
     (Password, with_options::Secret)
 );
-
-impl TryFrom<CsrConnectionOptionExtracted> for mz_storage::types::connections::CsrConnection {
-    type Error = PlanError;
-    fn try_from(ccsr_options: CsrConnectionOptionExtracted) -> Result<Self, Self::Error> {
-        let url = match ccsr_options.url {
-            Some(url) => url
+impl CsrConnectionOptionExtracted {
+    fn to_connection(
+        self,
+        scx: &StatementContext,
+    ) -> Result<mz_storage::types::connections::CsrConnection, PlanError> {
+        let url = match (self.url, self.aws_privatelink, self.port) {
+            (Some(url), None, None) => url
                 .parse()
                 .map_err(|e| sql_err!("parsing schema registry url: {e}"))?,
-            None => sql_bail!("invalid CONNECTION: must specify URL"),
+            (Some(_), None, Some(_)) => {
+                sql_bail!("invalid connection: cannot specify both URL and PORT options");
+            }
+            (None, Some(aws_privatelink), port) => {
+                let id = GlobalId::from(aws_privatelink);
+                let entry = scx.catalog.get_item(&id);
+                let port = port.unwrap_or(8081);
+                match entry.connection()? {
+                    // TODO: don't hardcode connection hostname logic.
+                    // TODO: introduce some way to specify the scheme (HTTP or HTTPS).
+                    Connection::AwsPrivatelink(_) => format!("https://connection-{id}:{port}")
```

Comment:
Where do we specify the FQDN to use for certificate validation? The `connection-{id}` will definitely not match the cert. It's probably fine to use the broker hostname they pass in, but I don't see where we specify that.

Reply:
This should be fixed now, by using [`reqwest::ClientBuilder::resolve_to_addrs`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.resolve_to_addrs) to rewrite the HTTP requests to hit the PrivateLink service without changing the domain name. Though I haven't tested.